### DISCUSSION TITLE
@Lazy is not respected in resolvers

### DISCUSSION THREAD

**Discussion by asfernandes** [graphql-java-kickstart/graphql-spring-boot#466] [Q&A]
**Describe the bug**
If I declare a `@Component @Lazy` resolver (`extends GraphQLQueryResolver`), its constructor is called in application startup and them all beans dependencies of this resolver are also resolved.

**To Reproduce**
Declare a `GraphQLQueryResolver` with `@Lazy`.

**Expected behavior**
Should work as others lazy beans.

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Java 8
 - graphql-spring-boot-starter 5.10.0

**Comment by oliemansm** [comment]
You're on a very old version of graphql-spring-boot-starter. We released version 8.0.0 today! Update that version too.

The starter does not create the beans themselves, the Spring ecosystem is responsible for creating these beans as you've defined them, so don't see how this is caused by this library.

**Comment by asfernandes** [comment]
My project is large and currently has incompatibilities with new versions of this library and graphql library which I was not able to resolve yet.

The Spring alone does respect `@Lazy`. Note that there is an `.graphqls` file referencing the resolver and seems the problem come from this relation.

**Comment by oliemansm** [comment]
Knowing Spring I'd advise to try to move away from those `@Lazy` annotations entirely, or put a layer in between somewhere to get more control. The fact that you're experiencing it using this library in your large project, doesn't mean it's caused by this library.

**Comment by asfernandes** [comment]
I do not use `@Lazy`, but I'm looking to use as startup is becoming slower and slower as I'm adding graphql resolvers.

Sorry if it's not problem of this library. The architecture of it is not very clear for me of what is its dependencies and what is the job of each one.

**Comment by asfernandes** [comment]
In the stack trace, it's `graphqlSchema` who is creating my lazy beans.
